### PR TITLE
Added view-element-admin as git submodule

### PR DESCRIPTION
**Pull request by EmmanuelObua**
- New file .gitmodules added to track git submodues.

### CODE REVIEW

**Comment by fregante** [comment]
Great! The submodule seems to be working correctly on vercel: https://playground-e73bqavr4-pixiebrix.vercel.app/vue-element-admin

However it needs to be built, so can you follow the instructions here?  Steps 2, 3, 4 https://github.com/pixiebrix/playground#adding-a-project-with-a-build

**Comment by EmmanuelObua** [comment]
I am reviewing

**Comment by fregante** [comment]
If you want to keep using the submodule, perhaps you can keep it under `/view-element-admin/repo` and then created `/view-element-admin/build.sh` to run commands inside it.

**Comment by twschiller** [comment]
> If you want to keep using the submodule, perhaps you can keep it under /view-element-admin/repo and then create /view-element-admin/build.sh to run commands inside it.

It would be good to keep using the submodule (because I don't think we will be modifying the project). @EmmanuelObua did you see a reason to include all the code directly? 

I'm not sure if you can see the build error? This is the error from the latest build:

```
[5/5] Building fresh packages...
--
09:38:00.690 | success Saved lockfile.
09:38:00.697 | Done in 53.50s.
09:38:00.975 | yarn run v1.22.17
09:38:01.045 | $ vue-cli-service build
09:38:01.858 |  
09:38:01.859 | -  Building for production...
09:38:45.175 | ERROR  Failed to compile with 5 errors2:38:45 PM
09:38:45.176 |  
09:38:45.176 | These dependencies were not found:
09:38:45.177 |  
09:38:45.177 | * core-js/modules/es.regexp.dot-all.js in ./mock/index.js, ./node_modules/cache-loader/dist/cjs.js??ref--12-0!./node_modules/thread-loader/dist/cjs.js!./node_modules/babel-loader/lib!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/components/ThemePicker/index.vue?vue&type=script&lang=js& and 1 other
09:38:45.177 | * core-js/modules/es.typed-array.at.js in ./src/components/ImageCropper/utils/data2blob.js, ./src/vendor/Export2Excel.js
09:38:45.177 |  
09:38:45.177 | To install them, you can run: npm install --save core-js/modules/es.regexp.dot-all.js core-js/modules/es.typed-array.at.js
09:38:45.178 | ERROR  Build failed with errors.
09:38:45.226 | error Command failed with exit code 1.
09:38:45.226 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
09:38:45.245 | Error: Command "bash ./build.sh" exited with 1
```

**Comment by fregante** [comment]
I'm not sure if Yarn is the issue, but I'd try using npm install in the install.sh script, we shouldn't need to fix their build.

**Comment by EmmanuelObua** [comment]
> > If you want to keep using the submodule, perhaps you can keep it under /view-element-admin/repo and then create /view-element-admin/build.sh to run commands inside it.
> 
> It would be good to keep using the submodule (because I don't think we will be modifying the project). @EmmanuelObua did you see a reason to include all the code directly?
> 
> I'm not sure if you can see the build error? This is the error from the latest build:
> 
> ```
> [5/5] Building fresh packages...
> --
> 09:38:00.690 | success Saved lockfile.
> 09:38:00.697 | Done in 53.50s.
> 09:38:00.975 | yarn run v1.22.17
> 09:38:01.045 | $ vue-cli-service build
> 09:38:01.858 |  
> 09:38:01.859 | -  Building for production...
> 09:38:45.175 | ERROR  Failed to compile with 5 errors2:38:45 PM
> 09:38:45.176 |  
> 09:38:45.176 | These dependencies were not found:
> 09:38:45.177 |  
> 09:38:45.177 | * core-js/modules/es.regexp.dot-all.js in ./mock/index.js, ./node_modules/cache-loader/dist/cjs.js??ref--12-0!./node_modules/thread-loader/dist/cjs.js!./node_modules/babel-loader/lib!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/components/ThemePicker/index.vue?vue&type=script&lang=js& and 1 other
> 09:38:45.177 | * core-js/modules/es.typed-array.at.js in ./src/components/ImageCropper/utils/data2blob.js, ./src/vendor/Export2Excel.js
> 09:38:45.177 |  
> 09:38:45.177 | To install them, you can run: npm install --save core-js/modules/es.regexp.dot-all.js core-js/modules/es.typed-array.at.js
> 09:38:45.178 | ERROR  Build failed with errors.
> 09:38:45.226 | error Command failed with exit code 1.
> 09:38:45.226 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> 09:38:45.245 | Error: Command "bash ./build.sh" exited with 1
> ```

I will take it back to git submodule then

**Comment by EmmanuelObua** [comment]
@twschiller , i am not able to see the build errors and it's making it abit hard to debug

**Comment by twschiller** [comment]
@EmmanuelObua I slacked you an invite to Vercel to view the build errors. 

It looks you might not be passing a command to npm? You should be able to run `build.sh` locally to determine if the build script works locally?

Let's also switch back to using a git submodule for the code

```
Usage: npm <command>
--
07:13:32.973 |  
07:13:32.973 | where <command> is one of:
07:13:32.973 | access, adduser, audit, bin, bugs, c, cache, ci, cit,
07:13:32.973 | clean-install, clean-install-test, completion, config,
07:13:32.974 | create, ddp, dedupe, deprecate, dist-tag, docs, doctor,
07:13:32.974 | edit, explore, fund, get, help, help-search, hook, i, init,
07:13:32.974 | install, install-ci-test, install-test, it, link, list, ln,
07:13:32.974 | login, logout, ls, org, outdated, owner, pack, ping, prefix,
07:13:32.974 | profile, prune, publish, rb, rebuild, repo, restart, root,
07:13:32.974 | run, run-script, s, se, search, set, shrinkwrap, star,
07:13:32.974 | stars, start, stop, t, team, test, token, tst, un,
07:13:32.975 | uninstall, unpublish, unstar, up, update, v, version, view,
07:13:32.975 | whoami
07:13:32.975 |  
07:13:32.975 | npm <command> -h  quick help on <command>
07:13:32.975 | npm -l            display full usage info
07:13:32.975 | npm help <term>   search for help on <term>
07:13:32.976 | npm help npm      involved overview
07:13:32.976 |  
07:13:32.976 | Specify configs in the ini-formatted file:
07:13:32.976 | /vercel/.npmrc
07:13:32.976 | or on the command line via: npm <command> --key value
07:13:32.976 | Config info can be viewed via: npm help config
07:13:32.976 |  
07:13:32.976 | npm@6.14.15 /node14/lib/node_modules/npm
07:13:32.992 | Error: Command "bash ./build.sh" exited with 1
```

File: vue-element-admin/public/.htaccess [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<ifModule mod_rewrite.c>
+    RewriteEngine On
+  RewriteBase /
+  RewriteRule ^index\.html$ - [L]
+  RewriteCond %{REQUEST_FILENAME} !-f
+  RewriteCond %{REQUEST_FILENAME} !-d
+  RewriteRule . /index.html [L]
+</ifModule>
```
Comment (fregante) [inline vue-element-admin/public/.htaccess:8]: .htaccess doesn't work here, but you can follow step 3 in the readme: https://github.com/pixiebrix/playground#adding-a-project-with-a-build

Which means you can just duplicate this line to achieve the same:

https://github.com/pixiebrix/playground/blob/703ce4b5e209a26b2eca851b4575398d69414d36/vercel.json#L6-L9

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "vue-element-admin"]
+	path = vue-element-admin
```
Comment (fregante) [inline .gitmodules:2]: `git subtree` still imports the whole repo into this one so it isn't beneficial to us. This PR currently adds about 300 files but it should only change 3 or 4. I think that what we're trying to do is achievable by changing this line to:

```suggestion
	path = vue-element-admin/repo
```

Then you'll be able to create the regular `/vue-element-admin/install.sh` and make the required changes in `/vercel.json`

**Pull request closed without merging**